packet_pool.go

```diff
@@ -5,17 +5,74 @@ import (
 	"sync"
 )
 
-// packetPool represents a pool of packets
+// packetAccumulator keeps track of packets for a single PID and decides when to flush them
+type packetAccumulator struct {
+	pid  uint16
+	q    []*Packet
+	pool *packetPool
+}
+
+// newPacketAccumulator creates a new packet queue for a single PID
+func newPacketAccumulator(pool *packetPool, pid uint16) *packetAccumulator {
```

Comment:
As for `packetPool`, I feel it would be better to inject both `parser` and `programMap` directly here instead of the `packetPool` itself

Reply:
:+1: